DeviceAdministration/Infrastructure/Repository/DeviceTelemetryRepository.cs

```diff
@@ -263,45 +263,77 @@ public DeviceTelemetryRepository(IConfigurationProvider configProvider)
                         model.DeviceId = str;
                     }
 
-                    if (strdict.TryGetValue("externaltemperature", out str) &&
-                        double.TryParse(
-                            str,
-                            NumberStyles.Float,
+                    DateTime date;
+                    if (strdict.TryGetValue("eventenqueuedutctime", out str) &&
+                        DateTime.TryParse(
+                            str, 
                             CultureInfo.InvariantCulture,
-                            out number))
+                            DateTimeStyles.AllowWhiteSpaces,
+                            out date))
                     {
-                        model.ExternalTemperature = number;
+                        model.Timestamp = date;
```

Comment:
Check what the default value of model.Timestamp and what happens if it isn't there.

If that's undefined behaviour, then we should allow it throw the error.